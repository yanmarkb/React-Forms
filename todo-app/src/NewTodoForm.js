import React, { useState } from "react";

const NewTodoForm = ({ addTodo }) => {
	const [formData, setFormData] = useState({
		task: "",
	});

	const handleChange = (evt) => {
		const { name, value } = evt.target;
		setFormData((data) => ({
			...data,
			[name]: value,
		}));
	};

	const handleSubmit = (evt) => {
		evt.preventDefault();
		addTodo({
			task: formData.task,
		});
		setFormData({
			task: "",
		});
	};

	return (
		<form onSubmit={handleSubmit}>
			<label htmlFor="task">Task:</label>
			<input
				id="task"
				type="text"
				name="task"
				value={formData.task}
				onChange={handleChange}
			/>
			<button type="submit">Add Todo</button>
		</form>
	);
};

export default NewTodoForm;
