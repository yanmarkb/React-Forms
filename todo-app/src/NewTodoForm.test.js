import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

test("NewTodoForm component renders without crashing", () => {
	render(<NewTodoForm addTodo={() => {}} />);
});

test("NewTodoForm component matches snapshot", () => {
	const { asFragment } = render(<NewTodoForm addTodo={() => {}} />);
	expect(asFragment()).toMatchSnapshot();
});

test("NewTodoForm component can add a new todo", () => {
	const addTodo = jest.fn();
	const { getByLabelText, getByText } = render(
		<NewTodoForm addTodo={addTodo} />
	);

	fireEvent.change(getByLabelText("Task:"), { target: { value: "New Task" } });
	fireEvent.click(getByText("Add Todo"));

	expect(addTodo).toHaveBeenCalledWith({
		task: "New Task",
	});
});
