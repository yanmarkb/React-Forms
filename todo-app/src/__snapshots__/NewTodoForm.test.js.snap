// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm component matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Task:
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
